index.js

```diff
@@ -221,6 +226,7 @@ class ReadableState {
 
   pipe (pipeTo, cb) {
     if (this.pipeTo !== null) throw new Error('Can only pipe to one destination')
+    if (this.error) throw new Error(`Stream already error: ${this.error}`)
```

Comment:
what's this guarding for?

Reply:
Will add some documentation/test explanation for this case. I got a good reason for it last night, but it elludes me atm.